p2p/pre2p/module.go

```diff
@@ -68,25 +70,51 @@ func (m *p2pModule) SetBus(bus modules.Bus) {
 
 func (m *p2pModule) GetBus() modules.Bus {
 	if m.bus == nil {
-		log.Fatalf("PocketBus is not initialized")
+		log.Printf("[WARN]: PocketBus is not initialized")
+		return nil
 	}
 	return m.bus
 }
 
 func (m *p2pModule) Start() error {
 	log.Println("Starting network module")
 
+	// TODO(derrandz): find a way to reset gauges at new consensus height
+	m.
+		GetBus().
+		GetTelemetryModule().
+		RegisterCounter(
+			"blockchain_nodes_connected",
+			"the counter to track the number of nodes online",
+		)
+
+	m.
+		GetBus().
+		GetTelemetryModule().
+		RegisterGauge("p2p_msg_broadcast_received_total_per_block", "the counter to track the number of messages received per block")
+
+	m.network.SetBus(m.GetBus())
+
 	go func() {
 		for {
 			data, err := m.listener.Read()
 			if err != nil {
 				log.Println("Error reading data from connection: ", err)
 				continue
 			}
+
 			go m.handleNetworkMessage(data)
 		}
 	}()
 
+	// TODO(tema):
+	// No way to know if the node has actually properly started as of the current implementation of m.listener
+	// Make sure to do this after the node has started if the implementation allowed for this in the future.
+	m.
+		GetBus().
+		GetTelemetryModule().
+		IncCounter("blockchain_nodes_connected")
```

Comment:
It is considered a decent practice to maintain gauge with a value of 1 that includes metadata in labels. E.g. port the service is running on can be a label in this case.

The name should be different though! The metric should carry a name from the node point of view, not from external point of view. So I'd suggest `p2p_listening`. Lmk if that doesn't make sense.

Reply:
I think this makes sense @okdas. Will include these changes in the next iteration.